providers/google/write.go

```diff
@@ -0,0 +1,102 @@
+package google
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/spyzhov/ajson"
+)
+
+func (c *Connector) Write(ctx context.Context, config common.WriteParams) (*common.WriteResult, error) { // nolint: cyclop,lll
+	if err := config.ValidateParams(); err != nil {
+		return nil, err
+	}
+
+	url, err := c.getURL(config.ObjectName)
+	if err != nil {
+		return nil, err
+	}
+
+	var write common.WriteMethod
+
+	if len(config.RecordId) == 0 {
+		if supportedObjectsByCreate[c.Module.ID].Has(config.ObjectName) {
+			write = c.Client.Post
+		}
+	} else {
+		if supportedObjectsByUpdate[c.Module.ID].Has(config.ObjectName) {
+			write = c.Client.Put
+
+			url.AddPath(config.RecordId)
+		}
+	}
+
+	if write == nil {
+		// No supported REST operation was found for current object.
+		return nil, common.ErrOperationNotSupportedForObject
+	}
+
+	if config.ObjectName == objectNameCalendarList {
```

Comment:
Nothing to do here, but can we automate this if we have a spec with the required fields?